src/lib/util/RunOutputProcessor.ts

```diff
@@ -52,6 +53,8 @@ export class RunOutputProcessor {
 		const msg = msgComponents.join('\n');
 		if (minSev > 0 && this.opts.violationsCauseException) {
 			throw new SfdxError(msg, null, null, minSev);
+		} else if (this.opts.severityForError && minSev <= this.opts.severityForError) {
+			throw new SfdxError(msg, null, null, minSev);
```

Comment:
We want the scanner to give a non-zero exit code only when the highest severity seen is equal to or more severe than the flags input value.

Reply:
Can you add this as a comment in the code?